File: bindings/dotnet/src/Zlink/Contracts/Sockets/RouterRoute.cs
// SPDX-License-Identifier: MPL-2.0

namespace Systems.Zlink;

/// <summary>
///     One row of a ROUTER selected-route snapshot: the routing id and the
///     generation of the route Core currently selects for it.
/// </summary>
/// <param name="RoutingId">Peer routing id.</param>
/// <param name="RouteGeneration">
///     Nonzero opaque equality token. It changes whenever Core selects a
///     different route for the same routing id; do not order values.
/// </param>
public readonly record struct RouterRoute(RoutingId RoutingId, ulong RouteGeneration);

File: bindings/java/src/main/java/systems/zlink/contracts/sockets/RoutedSocketContracts/RouterRoute.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.contracts.sockets;

import java.util.Objects;
import systems.zlink.contracts.core.RoutingId;

/**
 * One row of a ROUTER selected-route snapshot: the peer routing id and the
 * generation of the route Core currently selects for it.
 *
 * @param routingId peer routing id
 * @param routeGeneration nonzero opaque equality token; it changes whenever
 *     Core selects a different route for the same routing id. Compare it only
 *     for equality (it is an unsigned 64-bit value; do not order values).
 */
public record RouterRoute(RoutingId routingId, long routeGeneration) {
    public RouterRoute {
        Objects.requireNonNull(routingId, "routingId");
    }
}
